### PR TITLE
Add docs for user-defined composite types

### PR DESCRIPTION
**Pull request by rmloveland**
Fixes DOC-6380, DOC-6739

### CODE REVIEW

**Comment by rmloveland** [comment]
@jordanlewis requesting your review b/c you wrote https://github.com/cockroachdb/cockroach/pull/90491
however let me know if bandwidth issues and I can ask around on schema team for a review

File: v23.1/create-type.md [hunk 1]
```diff
@@ -25,32 +26,54 @@ Parameter | Description
...
 `IF NOT EXISTS` |  Create a new type only if a type of the same name does not already exist in the database; if one does exist, do not return an error.
-`opt_enum_val_list` | A list of values that make up the type's enumerated set.
+`opt_enum_val_list` | A list of values that make up the type's [enumerated set](#create-an-enumerated-data-type).
+`opt_composite_type_list` | A list of values that make up the set of types that make up a [composite type](#create-a-composite-data-type).
 
 ## Required privileges
 
-- To create a type, the user must have the `CREATE` [privilege](security-reference/authorization.html#managing-privileges) on the parent database.
-- To use a user-defined type in a table (e.g., when defining a column's type), the user must have the `USAGE` privilege on the type.
+- To create a type, the user must have [the `CREATE` privilege](security-reference/authorization.html#supported-privileges) on the parent database.
```
Comment (rafiss) [inline v23.1/create-type.md:34]: the user also needs to have the CREATE on the schema in which the type is being created.

File: v23.1/show-grants.md [hunk 2]
```diff
@@ -292,6 +292,8 @@ To list all grants for all users and roles on the current database and its table
 
 ### Show grants on user-defined types
 
+To show grant privileges on [user-defined types](create-type.html), use the following statements.
```
Comment (rafiss) [inline v23.1/show-grants.md:295]: nit: it could say "to show privileges"

**Review by rafiss — APPROVED** [review]
lgtm! just had two nits

File: v23.1/create-type.md (hunk 1 above)
Comment (rmloveland) [reply v23.1/create-type.md]: updated, thanks!

**Review by ianjevans — APPROVED** [review]

File: v23.1/create-type.md [hunk 3]
```diff
@@ -108,11 +131,76 @@ Parameter | Description
...
+Insert 10,000 randomly generated values of type `mypoint`:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+INSERT INTO points (p) SELECT (floor(random()*10000),floor(random()*10000))::MYPOINT FROM generate_series(1,10000);
+~~~
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+SELECT * FROM points ORDER BY ((p).x) ASC LIMIT 25;
```
Comment (dikshant) [inline v23.1/create-type.md:164]: There are some syntax errors in the example. Here is what it should look like:

```sql
CREATE TYPE my_point AS (id INT, x INT, y INT);

CREATE TABLE points (
    id UUID DEFAULT gen_random_uuid(),
    p my_point,
    CONSTRAINT mypoint_xcheck CHECK (((p).x >= 0) AND ((p).x <= 10000)),
    CONSTRAINT mypoint_ycheck CHECK (((p).y >= 0) AND ((p).y <= 10000))
);

-- Insert 10,000 randomly generated values of type `mypoint`:
INSERT INTO points (p) SELECT (floor(random()*10000),floor(random()*10000), floor(random()*10000))::MY_POINT FROM generate_series(1,10000);
SELECT * FROM points ORDER BY ((p).x) ASC LIMIT 25;
```

**Review by dikshant — CHANGES_REQUESTED** [review]
We will wanna fix that example above before merging.

File: v23.1/create-type.md [hunk 4]
```diff
@@ -5,13 +5,14 @@ toc: true
 docs_area: reference.sql
```
Comment (ianjevans) [inline v23.1/create-type.md:5]: Just noticed that the page summary needs to be updated.

File: v23.1/drop-type.md [hunk 5]
```diff
@@ -33,7 +33,7 @@ The user must be the owner of the type.
 - You cannot drop a type or view that is in use by a table.
```
Comment (ianjevans) [inline v23.1/drop-type.md:33]: Page summary needs to be updated here, too.

**Review by ianjevans — CHANGES_REQUESTED** [review]

File: v23.1/drop-type.md (hunk 5 above)
Comment (rmloveland) [reply v23.1/drop-type.md:33]: Updated

File: v23.1/create-type.md (hunk 4 above)
Comment (rmloveland) [reply v23.1/create-type.md:5]: Updated!

File: v23.1/create-type.md (hunk 3 above)
Comment (rmloveland) [reply v23.1/create-type.md]: Fixed the example, thanks for the catch!  Also ran it through a little SQL checker script I've been working on as a side project, so it should be good now. PTAL

**Review by ianjevans — APPROVED** [review]

**Pull request merged**